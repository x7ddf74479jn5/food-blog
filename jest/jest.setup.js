// Optional: configure or set up a testing framework before each test.
// If you delete this file, remove `setupFilesAfterEnv` from `jest.config.js`

// Used for __tests__/testing-library.js
// Learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";

window.matchMedia = jest.fn().mockImplementation((query) => {
  return {
    addListener: jest.fn(),
    matches: false,
    media: query,
    onchange: null,
    removeListener: jest.fn(),
  };
});
