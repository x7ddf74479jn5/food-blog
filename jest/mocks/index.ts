export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
