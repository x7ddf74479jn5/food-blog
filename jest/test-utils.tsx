import type Queries from "@testing-library/dom/types/queries";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";
import { SWRConfig } from "swr";

import { SearchProvider } from "@/contexts/search/SearchContext";
import { ViewportProvider } from "@/contexts/viewport/ViewportContext";

import { withMockLegacyRouter, withMockRouter } from "./mocks";

export const Providers: React.ComponentType<{ children?: React.ReactNode }> = ({ children }) => {
  return withMockRouter(
    withMockLegacyRouter(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <SearchProvider>
          <ViewportProvider>{children}</ViewportProvider>
        </SearchProvider>
      </SWRConfig>
    )
  );
};

const customRender = (ui: React.ReactElement, options = {}): RenderResult<typeof Queries, HTMLElement> => {
  return render(ui, { wrapper: Providers, ...options });
};

// re-export everything
export * from "./mocks";
export * from "@testing-library/react";

// override render method
export { render as originalRender, customRender as render };

export const reTestCase = {
  anyImage: expect.stringMatching(/^(data:image\/gif)|\.(png|webp|jpeg|jpg|svg)$/),
  anyWord: expect.stringMatching(/\w+/),
};

type ErrorComponentProps = {
  message?: string;
};
export const ErrorComponent: React.FC<ErrorComponentProps> = ({ message }) => {
  throw new Error(message);
};

export const setupMockServer = (...handlers: RequestHandler[]) => {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
};
