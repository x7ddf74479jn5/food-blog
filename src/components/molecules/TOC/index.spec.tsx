import { render, screen } from "jest/test-utils";
import React from "react";

import { TOC } from ".";

jest.mock("@/hooks/useWindowSize", () => {
  const useWindowSize = jest.fn().mockReturnValue({ width: 2000, height: 2000 });
  return useWindowSize;
});

describe("components/molecules/TOC", () => {
  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
    <div>
      <div id="js-toc-content" className="hidden">
        <h2 data-testid="heading">h2</h2>
        <h3 data-testid="heading">h3</h3>
        <h4 data-testid="heading">h4</h4>
        <h5 data-testid="heading">h5</h5>
        <h6 data-testid="heading">h6</h6>
      </div>
      {children}
    </div>
  );
  it("snapshot", () => {
    const { asFragment } = render(
      <Wrapper>
        <TOC />
      </Wrapper>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("OK: 初期レンダリング ", () => {
    render(
      <Wrapper>
        <TOC />
      </Wrapper>
    );

    const contentHeader = screen.getAllByRole("heading", { level: 2 })[1];
    expect(contentHeader).toHaveTextContent("目次");
    const anchors = screen.getAllByRole("link");
    const headings = screen.getAllByTestId("heading");
    for (let index = 0; index < anchors.length; index++) {
      expect(headings[index].innerText).toBe(anchors[index].innerText);
    }
  });
});
