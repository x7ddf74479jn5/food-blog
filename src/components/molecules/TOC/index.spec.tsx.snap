// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/TOC snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="hidden"
      id="js-toc-content"
    >
      <h2>
        h2
      </h2>
      <h3>
        h3
      </h3>
      <h4>
        h4
      </h4>
      <h5>
        h5
      </h5>
      <h6>
        h6
      </h6>
    </div>
    <section
      class="container p-2 dark:text-white bg-gray-50 dark:bg-gray-700"
    >
      <h2
        class="px-2 pb-2 mb-2 border-b-2"
      >
        目次
      </h2>
      <div
        class="toc"
        id="js-toc"
      >
        <ul
          class="toc-list "
        >
          <li
            class="toc-list-item"
          >
            <a
              class="toc-link node-name--H2 "
              href="#"
            >
              h2
            </a>
            <ul
              class="toc-list  is-collapsible"
            >
              <li
                class="toc-list-item"
              >
                <a
                  class="toc-link node-name--H3 "
                  href="#"
                >
                  h3
                </a>
                <ul
                  class="toc-list  is-collapsible"
                >
                  <li
                    class="toc-list-item"
                  >
                    <a
                      class="toc-link node-name--H4 "
                      href="#"
                    >
                      h4
                    </a>
                    <ul
                      class="toc-list  is-collapsible"
                    >
                      <li
                        class="toc-list-item"
                      >
                        <a
                          class="toc-link node-name--H5 "
                          href="#"
                        >
                          h5
                        </a>
                        <ul
                          class="toc-list  is-collapsible"
                        >
                          <li
                            class="toc-list-item is-active-li"
                          >
                            <a
                              class="toc-link node-name--H6  is-active-link"
                              href="#"
                            >
                              h6
                            </a>
                          </li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
